source/extensions/filters/http/jwt_authn/verify_context.cc

```diff
@@ -0,0 +1,53 @@
+#include "extensions/filters/http/jwt_authn/verify_context.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace JwtAuthn {
+namespace {
+
+class VerifyContextImpl : public VerifyContext {
+public:
+  VerifyContextImpl(Http::HeaderMap& headers, VerifierCallbacks* callback)
+      : headers_(headers), callback_(callback) {}
+
+  Http::HeaderMap& headers() const override { return headers_; }
+
+  VerifierCallbacks* callback() const override { return callback_; }
+
+  void responded(void* elem) override { responded_set_.insert(elem); }
```

Comment:
You may want to define a separate class to handle these Responded() stuff.   It will be easier to read.

Reply:
I'll try to move then to a separate class later.